Spigot-API-Patches/0017-API-for-Name-to-UUID-caching.patch

```diff
@@ -0,0 +1,66 @@
+From 91cec68f3e2a2bed046554b9549f2cd20de94321 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Fabian=20Fa=C3=9Fbender?=
+ <fabian.fassbender42@googlemail.com>
+Date: Wed, 3 Feb 2016 19:32:10 +0100
+Subject: [PATCH] API for Name to UUID caching
+
+
+diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
+index ae75bd4..a83f3de 100644
+--- a/src/main/java/org/bukkit/Server.java
++++ b/src/main/java/org/bukkit/Server.java
+@@ -42,6 +42,7 @@ import org.bukkit.generator.ChunkGenerator;
+ 
+ import org.bukkit.inventory.ItemFactory;
+ import org.bukkit.inventory.meta.ItemMeta;
++import org.github.paperspigot.cache.ProfileCacheAdapter;
+ 
+ /**
+  * Represents a server implementation.
+@@ -978,6 +979,13 @@ public interface Server extends PluginMessageRecipient {
+             throw new UnsupportedOperationException( "Not supported yet." );
+         }
+         // PaperSpigot end
++
++        // PaperSpigot start - Add setProfileCacheAdapter method
++        public void setProfileCacheAdapter( ProfileCacheAdapter adapter )
++        {
++            throw new UnsupportedOperationException( "Not supported yet." );
++        }
++        // PaperSpigot end
+     }
+ 
+     Spigot spigot();
+diff --git a/src/main/java/org/github/paperspigot/cache/ProfileCacheAdapter.java b/src/main/java/org/github/paperspigot/cache/ProfileCacheAdapter.java
+new file mode 100644
+index 0000000..31ce477
+--- /dev/null
++++ b/src/main/java/org/github/paperspigot/cache/ProfileCacheAdapter.java
+@@ -0,0 +1,24 @@
++package org.github.paperspigot.cache;
++
++import java.util.UUID;
++
++/**
++ * Used to bind an external Cache implementation like Redis into the UserCache of MC
++ */
++public interface ProfileCacheAdapter {
++    /**
++     * Resolve a name to its uuid
++     *
++     * @param name of the player in any case (like given into {@link org.bukkit.inventory.meta.SkullMeta#setOwner(String)})
++     * @return the resolved uuid or null if the cache does not recognize this name
++     */
++    UUID resolve( String name );
```

Comment:
The use of the interface is internally and the only case i know is a Name => UUID resolve. A UUID => Name resolve would be nice for a open API like OfflinePlayer.getName()